src/nvim/hardcopy.c

```diff
@@ -455,8 +454,10 @@ static void prt_line_number(prt_settings_T *psettings, int page_line, linenr_T l
  */
 int prt_header_height(void)
 {
-  if (printer_opts[OPT_PRINT_HEADERHEIGHT].present)
-    return printer_opts[OPT_PRINT_HEADERHEIGHT].number;
+  if (printer_opts[OPT_PRINT_HEADERHEIGHT].present) {
+    assert(printer_opts[OPT_PRINT_HEADERHEIGHT].number <= INT_MAX);
```

Comment:
That's true, in principle. I can't carefully review the code now, to see if it has sense doing so. I'll check that tomorrow.